Source/LuaBridge/RefCountedPtr.h

```diff
@@ -159,13 +159,17 @@ class RefCountedPtr : private detail::RefCountedPtrBase
     */
     void reset()
     {
-        if (m_p != 0)
+        auto myCount_it = getRefCounts().find(m_p); // use find() to avoid looking up m_p more than once
+        if (myCount_it != getRefCounts().end())
```

Comment:
I would tend here to offensive programming because this situation shall not happen if this class is implemented properly.
`assert(myCount_it != getRefCounts().end());`

Reply:
Will do. Speaking of implementing properly, should the copy constructor and operator= functions be checking for self assignment? Because they don't appear to be currently.

Reply:
You are right the assign operators are both not correct. There are two issues there. I'm going to fix them.